plugin/builtin/suffixtransformer/SuffixTransformer.go

```diff
@@ -65,20 +62,17 @@ func (p *plugin) Transform(m resmap.ResMap) error {
 				continue
 			}
 			// TODO: move this test into the filter.
-			if smellsLikeANameChange(&fs) {
+			if fs.Path == "metadata/name" {
 				// "metadata/name" is the only field.
-				// this will add a prefix and a suffix
-				// to the resource even if those are
-				// empty
+				// this will add a suffix to the resource
+				// even if it is empty
 
-				r.AddNamePrefix(p.Prefix)
 				r.AddNameSuffix(p.Suffix)
-				if p.Prefix != "" || p.Suffix != "" {
+				if p.Suffix != "" {
 					r.StorePreviousId()
```

Comment:
Same comment here to leave the following comment right before `r.StorePreviousID()`:

```
// TODO: There are multiple transformers that can change a resource's name, and each makes a call to 
// StorePreviousID(). We should make it so that we only call StorePreviousID once per kustomization layer 
// to avoid storing intermediate names between transformations, to prevent intermediate name conflicts. 
```